### PR TITLE
Storybook v8 stories batch 7

### PR DESCRIPTION
**Pull request by misama-ct**
## Background
In an effort to switch from Storybook v5 to Storybook v8, the existing stories had to be migrated to a new storybook-format and converted to typescript. 

This pull-request is [part of a batch](https://github.com/commercetools/ui-kit/pull/2875). It contains the typescript-equivalents of existing storybook v5 stories and aims to replicate the same functionality / value. 

## Goal
Feature parity between v8 and v5 storybook. After merging all batches, a product developer who looked at the storybook v5 version yesterday, should be able to look at the v8 version tomorrow and be as productive (or more productive) as before. 

## Review instructions
A thorough **code review** is not required yet. The code was either copied from the existing js-equivalent or quickly written from scratch to replicate what was already there to fit the new story-format. It is expected that those stories contain typescript errors, introduced by the conversion or previously existing. (Another pull-request will take care of those issues at a later stage).

What is expect is a **story review**:

You will need to compare the v5 with the v8 storybook (ideally side-by-side) and make sure that every component in this pull-request has:

- [ ] a readme-file in the same or a parent-folder
- [ ] a props-table displaying available props (and appropriate inputs for them)
- [ ] one or more stories that showcase the same or more functionality as the v5 equivalent

You will find the links to the different storybooks below. 

> If something is missing or irritating, add a comment to the source file here in the pull-request to start a conversation.

### CODE REVIEW

**Comment by FilPob** [comment]
Stumbled across the content notification story. Per default the Type is not set. That leads to the notification being shown with a style that is not intended 
image

**Comment by ddouglasz** [comment]
**PrimaryActionDropDown**
I am missing the disabled and label properties for interacting with each option

image

**Comment by ddouglasz** [comment]
**WarningMessage**
clicking the intlMessage control breaks the story

image

**Comment by ddouglasz** [comment]
**ContentNotification**

1. Clicking `intlMessage` control also breaks the story
2. We are missing the conrtol for `onRemove`

**Comment by ddouglasz** [comment]
**Stamp**
Clicking the children control breaks the Story

**Comment by ddouglasz** [comment]
**ViewSwitcher**
Removing any of the items from the `children` control breaks the Story

**Comment by misama-ct** [comment]
> Stumbled across the content notification story. Per default the Type is not set. That leads to the notification being shown with a style that is not intended

@FilPob 

✅ Fixed

**Comment by misama-ct** [comment]
> **PrimaryActionDropDown** I am missing the disabled and label properties for interacting with each option
> 

@ddouglasz 
I didn't even attempt to replicate this. Even if it's somehow possible in sb v8 with some hacks, I don't think it adds value. The Basic example contains a disabled Item, the basic examples' code demonstrates every `Option` prop and the tab-bar below the example allows to show the prop-table for the `Option` component as well. 

image

**Comment by misama-ct** [comment]
> **WarningMessage** clicking the intlMessage control breaks the story
> 
@ddouglasz 

✅ I deactivated the `intlMessage` control.

**Comment by misama-ct** [comment]
> **Stamp**
> Clicking the children control breaks the Story

@ddouglasz 
✅  According to the `TStampProps` type, `children` is deprecated in favor of `label`. I added the `@deprecated` annotation to the `children` prop's  typescript description and removed the `control` for `children` from the table.

**Comment by misama-ct** [comment]
> **ContentNotification**
> 
> 1. Clicking `intlMessage` control also breaks the story
> 2. We are missing the conrtol for `onRemove`

@ddouglasz 
1. ✅ Fixed (I removed the control-input from the table)
2. There is the extra story _Dismissable Notification_ to demonstrate the `onRemove` prop (= if onRemove is defined, show the closing button)

**Comment by misama-ct** [comment]
> **ViewSwitcher**
> Removing any of the items from the `children` control breaks the Story

@ddouglasz 

I removed the `children` control from the props-table since the value needs to be a `ReacNode` and this can not be handled by any regular sb control-input. I also removed the `defaultSelected` control, because its' effect can not be demonstrated after the component has been mounted already.

**Review by ddouglasz — APPROVED** [review]
Thank you for the follow up! 🙌🏽

**Review by FilPob — APPROVED** [review]

File: packages/components/messages/src/error-message/error-message.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import ErrorMessage from './error-message';
+
+const meta: Meta<typeof ErrorMessage> = {
+  title: 'components/Messages/ErrorMessage',
+  component: ErrorMessage,
```
Comment (ByronDWall) [inline packages/components/messages/src/error-message/error-message.stories.tsx:6]: Like in `WarningMessage`, setting an `intlMessage` breaks the example component.
```suggestion
  component: ErrorMessage,
    argTypes: {
    intlMessage: { control: { disable: true } },
  },
```

**Review by ByronDWall — APPROVED** [review]
One small nit, but Im going to go ahead and approve.  The spacings in particular are a big improvement, thanks!

**Pull request merged**